spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageResourceObject.java

```diff
@@ -161,25 +188,33 @@ public File getFile() throws IOException {
 
 	@Override
 	public long contentLength() throws IOException {
-		return throwExceptionForNullBlob(getGoogleStorageObject()).getSize();
+		return throwExceptionForNullBlob(getGcsBlob()).getSize();
 	}
 
 	@Override
 	public long lastModified() throws IOException {
-		return throwExceptionForNullBlob(getGoogleStorageObject()).getUpdateTime();
+		return throwExceptionForNullBlob(getGcsBlob()).getUpdateTime();
 	}
 
 	@Override
-	public Resource createRelative(String relativePath) throws IOException {
+	public GoogleStorageResourceObject createRelative(String relativePath) throws IOException {
 		int lastSlashIndex = this.location.lastIndexOf("/");
 		String absolutePath = this.location.substring(0, lastSlashIndex + 1) + relativePath;
-		return new GoogleStorageResourceObject(this.storage, absolutePath, this.createBlobIfNotExists);
+
+		GoogleStorageResourceObject blobResource = new GoogleStorageResourceObject(
+				this.storage, absolutePath, this.createBlobIfNotExists);
```

Comment:
The javadoc for the method should mention that `createBlobIfNotExists` is passed up the bucket.

Reply:
That would be adding complexity.
The property is called `autoCreateFiles`. I think it conveys that files and buckets are automatically created when needed.
I'm good with the naming as-is, but if you want to rename to something that's more general and encompasses files and buckets, sure.

Reply:
OK fair enough. My opinion was always that buckets should never be auto-created. I'll create an issue for that.